agent/connect/ca/provider_vault_auth_jwt.go

```diff
@@ -0,0 +1,50 @@
+package ca
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+func NewJwtAuthClient(authMethod *structs.VaultAuthMethod) (*VaultAuthClient, error) {
+	params := authMethod.Params
+
+	role, ok := params["role"].(string)
+	if !ok || strings.TrimSpace(role) == "" {
+		return nil, fmt.Errorf("missing 'role' value")
+	}
+
+	authClient := NewVaultAPIAuthClient(authMethod, "")
+	// The path is required for the auto-auth config, but this auth provider
+	// seems to be used for jwt based auth by directly passing the jwt token.
+	// So we only require the token file path if the token string isn't
+	// present.
```

Comment:
This comment is a little confusing to read - is it referring to L28 below?

Reply:
Opps.. forgot to move that down when I switched the code over to use legacyCheck(). 
IE. you are right that this is about line 28.